src/coreclr/jit/fginline.cpp

```diff
@@ -1584,6 +1584,9 @@ void Compiler::fgInsertInlineeBlocks(InlineInfo* pInlineInfo)
 
     lvaGenericsContextInUse |= InlineeCompiler->lvaGenericsContextInUse;
 
+    // Caller has no "init locals" flag, but callee does? Propagate the flag up.
+    info.compInitMem |= InlineeCompiler->info.compInitMem;
+
```

Comment:
I don't think this is the right fix. We should instead zero this in when we do the transformation during import or when we detect the mismatch during inlining. Note that we already detect and handle the case during inlining for all other locals than the one introduced by this transformation.

The issue is the same as #95056.

Reply:
> I don't think this is the right fix. We should instead zero this in when we do the transformation during import or when we detect the mismatch during inlining. Note that we already detect and handle the case during inlining for all other locals than the one introduced by this transformation.
> 
> The issue is the same as #95056.

That was my first fix too, but I just wanted to see the diffs from this

Reply:
Also, I guess it's a correct fix, just likely too conservative.

Reply:
Maybe it's correct due to limitations around where we currently allow inlining the converted stackallocs, but in general initializing inlinee locals only in the prolog of the caller is not sufficient since the inlined callee can run multiple times... So it seems like a bit of a fragile fix.